lib/ansible/modules/debconf.py

```diff
@@ -132,22 +132,23 @@
 def get_password_value(module, pkg, question, vtype):
     getsel = module.get_bin_path('debconf-get-selections', True)
     cmd = [getsel]
-    rc, out, err = module.run_command(cmd)
+    rc, out, dummy = module.run_command(cmd)
     if rc != 0:
         module.fail_json(msg="Failed to get the value '%s' from '%s'" % (question, pkg))
 
     desired_line = None
     for line in out.split("\n"):
         if line.startswith(pkg):
-            desired_line = line
-            break
+            (dpkg, dquestion, dvtype, *dvalue) = line.split('\t')
+            if dquestion == question and dvtype == vtype:
+                desired_line = line
+                if len(dvalue) >= 1:
+                    return dvalue[0]
+                return ''
 
     if not desired_line:
```

Comment:
with the changes above, there should be no `if` here as desired line will never be not None at this point

same with the `return ''` , since you moved the logic into the for loop, it would never be hit